cartofacade/carto_facade_test.go

```diff
@@ -180,34 +180,35 @@ func TestStart(t *testing.T) {
 
 	cartoFacade := New(&lib, cfg, algoCfg)
 	carto := CartoMock{}
-	carto.StartFunc = func() error {
-		return nil
-	}
 	cartoFacade.carto = &carto
 	cartoFacade.startCGoroutine(cancelCtx, &activeBackgroundWorkers)
 
-	t.Run("testing Start", func(t *testing.T) {
-		// success case
+	t.Run("success", func(t *testing.T) {
+		carto.StartFunc = func() error {
+			return nil
+		}
+		cartoFacade.carto = &carto
```

Comment:
[q] Do I need to set `cartoFacade.carto = &carto` each time I overwrite `carto.StartFunc`? I realize now that it's probably not necessary. What do you think @jeremyrhyde @kim-mishra ?

Reply:
cartoFacade.carto = &carto is not necessary